lib/Service/SlackAPIService.php

```diff
@@ -118,21 +121,46 @@ private function getUserRealName(string $userId, string $slackUserId): ?string {
 
 	/**
 	 * @param string $userId
+	 * @param bool $useCache
 	 * @return array
 	 * @throws PreConditionNotMetException
 	 */
-	public function getMyChannels(string $userId): array {
-		$channelResult = $this->request($userId, 'conversations.list', [
-			'exclude_archived' => true,
-			'types' => 'public_channel,private_channel,im,mpim'
-		]);
-
-		if (isset($channelResult['error'])) {
-			return (array)$channelResult;
+	public function getMyChannels(string $userId, bool $useCache): array {
+		$dCache = $this->cacheFactory->createDistributed(Application::APP_ID);
+		$cacheKey = 'channels-' . $userId;
+
+		if (!$useCache) {
+			$dCache->remove($cacheKey);
```

Comment:
Why invalidating the cache if the channels are requested without using the cache?
The cached value will be overwritten later in this method anyway.

Maybe you wanna make sure the cached value is cleared if the request fails. I don't think that's necessary, the TTL is enough.

IMO, those 2 lines can be removed.